dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java

```diff
@@ -17,93 +17,62 @@
- * 234234,0,5,1234123
- * 345234,0,345234.345,3423
- * </code>
- * </pre>
- * Meanwhile, open another console to see the output:
- * <pre>
- * <code>
- * bin/kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic sub_orders
- * **Expected :
- * 123445,0,100.0
- * 345234,0,345234.345
- * </code>
- * </pre>
  */
-public class BeamSqlExample implements Serializable {
+public class BeamSqlExample {
+  private static final Logger LOG = LoggerFactory.getLogger(BeamSqlExample.class);
 
   public static void main(String[] args) throws Exception {
-    PipelineOptions options = PipelineOptionsFactory.fromArgs(new String[] {}).withValidation()
-        .as(PipelineOptions.class); // FlinkPipelineOptions.class
-    options.setJobName("BeamSqlExample");
-    Pipeline pipeline = Pipeline.create(options);
+    PipelineOptions options = PipelineOptionsFactory.create();
+    Pipeline p = Pipeline.create(options);
 
-    BeamSQLEnvironment runner = BeamSQLEnvironment.create();
-    runner.addTableMetadata("ORDER_DETAILS", getTable("127.0.0.1:9092", "orders"));
-    runner.addTableMetadata("SUB_ORDER", getTable("127.0.0.1:9092", "sub_orders"));
+    //define the input row format
+    BeamSQLRecordType type = new BeamSQLRecordType();
+    type.addField("c1", SqlTypeName.INTEGER);
+    type.addField("c2", SqlTypeName.VARCHAR);
+    type.addField("c3", SqlTypeName.DOUBLE);
+    BeamSQLRow row = new BeamSQLRow(type);
+    row.addField(0, 1);
+    row.addField(1, "row");
+    row.addField(2, 1.0);
 
-    // case 2: insert into <table>(<fields>) select STREAM <fields> from
-    // <table> from <clause>
-    String sql = "INSERT INTO SUB_ORDER(order_id, site_id, price) " + "SELECT "
-        + " order_id, site_id, price " + "FROM ORDER_DETAILS " + "WHERE SITE_ID = 0 and price > 20";
+    //create a source PCollection with Create.of();
+    PCollection<BeamSQLRow> inputTable = PBegin.in(p).apply(Create.of(row));
+    inputTable.setCoder(new BeamSqlRowCoder(type));
 
-    PCollection<BeamSQLRow> outputStream = runner.compileBeamPipeline(sql, pipeline);
+    //run a simple SQL query over input PCollection;
+    String sql = "select c2, c3 from TABLE_A where c1=1";
+    PCollection<BeamSQLRow> outputStream = inputTable.apply(BeamSql.simpleQuery(sql));
 
-    pipeline.run().waitUntilFinish();
-  }
-
-  public static BaseBeamTable getTable(String bootstrapServer, String topic) {
-    final RelProtoDataType protoRowType = new RelProtoDataType() {
+    //log out the output record;
+    outputStream.apply("log_result",
+        MapElements.<BeamSQLRow, Void>via(new SimpleFunction<BeamSQLRow, Void>() {
       @Override
-      public RelDataType apply(RelDataTypeFactory a0) {
-        return a0.builder().add("order_id", SqlTypeName.BIGINT).add("site_id", SqlTypeName.INTEGER)
-            .add("price", SqlTypeName.DOUBLE).add("order_time", SqlTypeName.TIMESTAMP).build();
+      public Void apply(BeamSQLRow input) {
+        LOG.info(input.valueInString());
```

Comment:
It would be best if toString() returned what a user wanted for display purposes instead of having `valueInString`